src/analyses/constant_propagator.cpp

```diff
@@ -269,10 +269,18 @@ bool constant_propagator_domaint::ai_simplify(
   exprt &condition,
   const namespacet &ns) const
 {
-  bool b=values.replace_const.replace(condition);
-  b&=simplify(condition, ns);
+  bool no_change = true;
 
-  return b;
+  while(!values.replace_const.replace(condition))
+  {
```

Comment:
Is there a test case for this?  As in; do you know where it is needed?

Reply:
That's needed when local constant propagation yields a subexpression that is not an lvalue, as in the following:
```
int x = 1;
int *p = &x;
x = *p;
```
Now the first substitution+simplification will be `x = x;` when only the second round will yield `x = 1;`.